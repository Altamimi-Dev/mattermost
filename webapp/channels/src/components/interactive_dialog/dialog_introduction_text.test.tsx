// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {renderWithContext} from 'tests/react_testing_utils';

import DialogIntroductionText from './dialog_introduction_text';

describe('components/DialogIntroductionText', () => {
    test('should render message with supported values', () => {
        const descriptor = {
            value: '**bold** *italic* [link](https://mattermost.com/) <br/> [link target blank](!https://mattermost.com/)',
        };
        const wrapper = renderWithContext(<DialogIntroductionText {...descriptor}/>);
        expect(wrapper.asFragment()).toMatchSnapshot();
    });

    test('should not fail on empty value', () => {
        const descriptor = {
            value: '',
        };
        const wrapper = renderWithContext(<DialogIntroductionText {...descriptor}/>);
        expect(wrapper.asFragment()).toMatchSnapshot();
    });
});
