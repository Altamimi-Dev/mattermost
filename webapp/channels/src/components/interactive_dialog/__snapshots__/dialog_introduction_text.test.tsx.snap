// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/DialogIntroductionText should not fail on empty value 1`] = `
<DocumentFragment>
  <span />
</DocumentFragment>
`;

exports[`components/DialogIntroductionText should render message with supported values 1`] = `
<DocumentFragment>
  <p>
    <strong>
      bold
    </strong>
     
    <em>
      italic
    </em>
     
    <a
      class="theme markdown__link"
      href="https://mattermost.com/"
      rel="noreferrer"
      target="_blank"
    >
      link
    </a>
     &lt;br/&gt; 
    <a
      class="theme markdown__link"
      href="!https://mattermost.com/"
      rel="noreferrer"
      target="_blank"
    >
      link target blank
    </a>
  </p>
</DocumentFragment>
`;
